datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -60,6 +60,151 @@ pub(crate) enum ExecutionState {
 
 use super::AggregateExec;
 
+/// An interning store for group keys
+trait GroupValues: Send {
```

Comment:
Eventually I would love to see this in its own module `aggregates/group_values.rs` but that can be a follow on PR for sure